Common++/src/IpAddress.cpp

```diff
@@ -178,4 +180,174 @@ namespace pcpp
 	}
 
 
+	// ~~~~~~~~~~~
+	// IPv4Network
+	// ~~~~~~~~~~~
+
+
+	bool IPv4Network::isValidSubnetMask(const std::string& subnetMask)
```

Comment:
I would make this belong to IPv4address

Reply:
I'm not sure what you mean... do you mean that `subnetMask` should be an `IPv4Address` and not a `std::string`?

Reply:
yes, that is correct. The reasons this method takes a string and not `IPv4Address` are:
- This is a private method so it's not exposed to the user
- The input we get from the user is a string, like [here](https://github.com/seladb/PcapPlusPlus/blob/4a5eb6f8d52215b490b92dfefde53365f6640491/Common%2B%2B/src/IpAddress.cpp#L247) and [here](https://github.com/seladb/PcapPlusPlus/blob/4a5eb6f8d52215b490b92dfefde53365f6640491/Common%2B%2B/src/IpAddress.cpp#L263)
- I thought about making it an `IPv4Address` but this might be a bit confusing because, as you mentioned, a subnet mask is a restricted IP address and for most users it's easier to just type "255.255.0.0"